gninasrc/lib/parallel_mc.cpp

```diff
@@ -37,6 +37,26 @@ struct parallel_mc_task
 	parallel_mc_task(const model& m_, int seed) :
 			m(m_), generator(static_cast<rng::result_type>(seed))
 	{
+        if (m_.gpu_initialized()) {
+            //TODO: need to ensure that worker threads using these copies can't
+            //deallocate GPU memory - race condition in
+            //parallel_mc_aux::operator(), plus inefficiency of having to copy
+            //buffers that won't change but could be freed
+            m.gdata.coords = m_.gdata.coords;

```

Comment:
m.gdata = m_.gdata  doesn't work?  If not, it probably should.  It's more appropriate for gdata to manage its members than parallel_mc_task